Directory.Build.targets

```diff
@@ -108,4 +108,11 @@
           BeforeTargets="Test"
           DependsOnTargets="GenerateCoverageResult"
           Condition="'$(IsUnitTestProject)' == 'true' AND '$(TargetFramework)' != '' AND '$(CollectCoverage)' == 'true'" />
+
+  <Target Name="ValidateNoTargetsProjectExtension"
+          BeforeTargets="PrepareForBuild"
+          Condition="'$(UsingMicrosoftNoTargetsSdk)' == 'true'">
+    <!-- Make sure any NoTargets projects use the proj extension so that they don't get language specific behavior from other SDKs -->
+    <Error Text="The project file should have a .proj extension since it uses the NoTargets SDK." Condition="'$(MSBuildProjectExtension)' != '.proj'" />
```

Comment:
Note that you need a language specific extension today for the correct multi-targeting behavior. In runtime we had to use the .csproj extension for that reason in a few projects: https://github.com/dotnet/runtime/blob/5acfe60b87fec6a47bd8ba891da4c6c5f3a04632/src/libraries/System.Composition/src/System.Composition.csproj#L1.

Reply:
I see - `ProjectReferences` don't work correctly when there is more than one `TargetFramework`.  What do you think it missing from NoTargets to support that?  Feels like all that should be part of the common targets.

Reply:
Unfortunately I don't have cycles to look into this right now but maybe @jeffkl knows more about this as he added support for language targets in NoTargets a while ago.

Reply:
Not asking to do it now, just wondering if that was some intentional exclusion or just work that wasn't done yet.

Reply:
The target framework negotiation logic is in language specific targets that only get imported if the project extension is `.csproj`. Importing them is more difficult than it sounds since there are a few places where the .NET SDK looks at the project extension and makes decisions.  

In my opinion, NoTargets projects shouldn't really multi-target since they're supposed to just copy files around or something simple.  But I'm sure there are advanced use cases in repos like this that warrant it.

Reply:
We aren't using it *yet* but dotnet/runtime is.  In that case they had a multi-targeting project representing framework-specific project references.

I could imagine us needing it if we were wanting to multi-target any of our redist packages (for example, Microsoft.ML) and get framework specific content from project references.  I guess for now I'll remove the enforcement of the file extension so that we can multi-target later if we want to.

Reply:
Then I would recommend the NoTargets projects that multi-target have an extension of `.csproj`.  They load in Visual Studio better and work better all around.